lib/compress.js

```diff
@@ -2611,7 +2611,8 @@ merge(Compressor.prototype, {
                 if (compressor.option("unused")
                     && def.references.length == 1
                     && compressor.find_parent(AST_Scope) === def.scope) {
-                    if (!compressor.option("keep_fnames")) {
+                    if (!compressor.option("keep_fnames")
+                        && exp.name && exp.name.definition() === def) {
                         exp.name = null;
```

Comment:
I think it's better to get this fix merged and a new release made. You can tweak it later.